exporter/datadogexporter/internal/hostmetadata/provider/provider.go

```diff
@@ -32,14 +34,24 @@ func (p *chainProvider) Source(ctx context.Context) (source.Source, error) {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
+	// Make a different context for our provider calls, to differentiate between a provider timing out and the entire
+	// context being cancelled
+	var childCtx context.Context
+	if p.timeout != 0 {
```

Comment:
I'm a little confused - which part should we move where?  Moving the creation of the child context into the `Chain` function?

Reply:
Moving the `WithTimeout` up until we call `Source` instead of here. It does mean adding it in more places, which is a bit more annoying. I'll leave what to do up to you

Also, I am now wondering if we want a separate timeout for this or if just [the per-OTLP blob timeout added by the exporterhelper](https://github.com/open-telemetry/opentelemetry-collector/blob/54b8c191dfd4d04073269a90943a55f430d90ed6/exporter/exporterhelper/timeout_sender.go#L47) works . Probably we want a separate one?

Reply:
Ah - I think its good to leave in here.  I think its good for this logic to be right next to the call site and if we ever want to do something fancier like per source timeout it will be much easier to add it here.

I definitely think we should want a separate timeout for this!  the per otlp blob timeout should be used just for the normal export case - not the occasions we're blocked on fetching a source.